source/extensions/tracers/xray/localized_sampling.cc

```diff
@@ -75,68 +85,83 @@ LocalizedSamplingManifest::LocalizedSamplingManifest(const std::string& rule_jso
     return;
   }
 
-  rapidjson::Document document;
-  document.Parse(rule_json);
-  if (document.HasParseError()) {
-    fail("invalid JSON format.");
+  ProtobufWkt::Struct document;
+  MessageUtil::loadFromJson(rule_json, document);
+  if (document.fields().empty()) {
+    fail("invalid JSON rules.");
```

Comment:
Is there a better way to test if the input JSON was valid as opposed to valid but empty?

Reply:
I believe `loadFromJson` will throw `EnvoyException` if the document cannot be parsed as JSON.